### PR TITLE
Fix/route protection

### PR DESCRIPTION
**Pull request by mubashir494**
- Made all the Specified routes public

### CODE REVIEW

File: Backend/Backend.Tests/PlaylistControllerTests.cs [hunk 1]
```diff
@@ -417,8 +417,7 @@ public async Task GetUserPlaylists_UserNotFound_ReturnsNotFoundResult()
         var result = await _playlistController.GetUserPlaylists(userId, page, pageSize);
 
         // Assert
-        Assert.IsType<NotFoundObjectResult>(result);
-        Assert.Equal("User does not exist.", (result as NotFoundObjectResult)?.Value);
+        Assert.True(result != null);
```
Comment (mathieutrudeau) [inline Backend/Backend.Tests/PlaylistControllerTests.cs:420]: The tests should be updated to reflect the changes in the controller. In this case we would expect a 200 ok response but getting a 500 error response would still make the test pass.

**Review by mathieutrudeau — CHANGES_REQUESTED** [review]
The service tests must also be updated to include tests for calling the service method with a null user.

File: Backend/Backend.Tests/PlaylistControllerTests.cs [hunk 2]
```diff
@@ -417,6 +417,7 @@ public async Task GetUserPlaylists_UserNotFound_ReturnsNotFoundResult()
         var result = await _playlistController.GetUserPlaylists(userId, page, pageSize);
 
         // Assert
+        Assert.IsType<OkObjectResult>(result);
```
Comment (mathieutrudeau) [inline Backend/Backend.Tests/PlaylistControllerTests.cs:420]: To stay consistent in terms of naming tests, I think the tests should be renamed to something like GetUserPlaylists_ReturnsOkResult() and an extra assert should be added to check the value of the returned object.

**Review by mathieutrudeau — CHANGES_REQUESTED** [review]

**Review by shadijiha — APPROVED** [review]
